apps/prefixSum/prefixSum-stdexec.cpp

```diff
@@ -0,0 +1,209 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2023 The Regents of the University of California,
+ * through Lawrence Berkeley National Laboratory (subject to receipt of any
+ * required approvals from the U.S. Dept. of Energy).All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*
+ * commons for the prefixSum codes
+ */
+
+#define PSUM_STDEXEC
+#include "prefixSum.hpp"
+#include "repeat_n/repeat_n.cuh"
+
+//
+// stdexec prefixSum function
+//
+template <typename T>
+[[nodiscard]] ull_t* prefixSum(scheduler auto &&sch, const T *in, const int N)
+{
+    ull_t *y = new ull_t[N+1];
+
+    // memcpy to output vector
+    ex::sync_wait(ex::schedule(sch) | ex::bulk(N, [=](int k){ y[k] = in[k]; }));
+
+    int niters = ilog2(N);
+    int *d_ptr = new int(0);
+
+    // GE Blelloch (1990) algorithm
+    // upsweep algorithm
+    ex::sender auto uSweep = ex::just()
+    | exec::on(sch,
+       repeat_n(niters, ex::bulk(N/2, [=](int k){
+            int d = *d_ptr;
+            int s1 = 1 << d+1;
+            int s2 = 1 << d;
+            int my = (k+1) * s1 - 1;
```

Comment:
please rename variables in a more expressive way, like `sum1`, or `currentSum` for `my` or anything easier to understand the algorithms.